disdrodb/data_transfer/download_data.py

```diff
@@ -0,0 +1,282 @@
+import os
+import re
+import zipfile
+import pooch
+import yaml
+
+import click
+import tqdm
+
+
+# function to check if a path exists
+def check_path(path: str) -> None:
+    """Check if a path exists
+
+    Parameters
+    ----------
+    path : str
+        Path to check
+
+    Raises
+    ------
+    FileNotFoundError
+        If the path does not exist
+    """
+    if not os.path.exists(path):
+        raise FileNotFoundError(f"The path {path} does not exist. Please check the path.")
+
+
+def get_metadata_folders(folder_path: str) -> list:
+    """Return a list containing all metadata folders paths
+
+    Parameters
+    ----------
+    folder_path : str
+        folder path
+
+    Returns
+    -------
+    list
+        list of metadata folders
+    """
+
+    return [os.path.join(root, "metadata") for root, dirs, files in os.walk(folder_path) if "metadata" in dirs]
+
+
+def get_yaml_files(metadata_folders: str) -> list:
+    """Return a list containing all yaml files paths
+
+    Parameters
+    ----------
+    metadata_folders : str
+        folder path
+
+    Returns
+    -------
+    list
+        list of yaml files
+    """
+
+    return [
+        os.path.join(root, file)
+        for metadata_folder in metadata_folders
+        for root, dirs, files in os.walk(metadata_folder)
+        for file in files
+        if file.endswith(".yml")
+    ]
+
+
+def get_list_urls_local_paths(
+    raw_folder: str = None,
+    data_source: str = None,
+    campaign_name: str = None,
+    station_name: str = None,
+) -> list:
+    """Parse the folder according to the paramerters (data_source,
+    campaign_name and station_name) to get all url from the config files.
+
+    Returns
+    -------
+    list
+        List of tuples containing the local path and the url.
+    """
+    # test if the file path is correct and exists. If not, raise an error
+    check_path(raw_folder)

```

Comment:
Use `_get_disdrodb_directory` https://github.com/ltelab/disdrodb/blob/main/disdrodb/api/io.py#L27 to get the `path_to_analyse`
Maybe add a `check_disdrodb_dir` function inside it to check `disdrodb_dir` ends with "DISDRODB"

Reply:
Or use

 list_stations_info = available_stations  (https://github.com/ltelab/disdrodb/blob/main/disdrodb/api/io.py#L340)
 
 metadata_dict = read_station_metadata(disdrodb_dir, product_level, data_source, campaign_name, station_name)  (https://github.com/ltelab/disdrodb/blob/main/disdrodb/api/metadata.py#L26)